ctapipe/tools/tests/test_train.py

```diff
@@ -0,0 +1,96 @@
+import pytest
+
+from ctapipe.core import run_tool
+from ctapipe.utils.datasets import resource_file
+
+
+def test_train_energy_regressor(energy_regressor_path):
+    from ctapipe.reco import EnergyRegressor
+
+    EnergyRegressor.read(energy_regressor_path)
+
+
+def test_train_particle_classifier(particle_classifier_path):
+    from ctapipe.reco import ParticleClassifier
+
+    ParticleClassifier.read(particle_classifier_path)
+
+
+def test_too_few_events(
+    model_tmp_path, dl2_shower_geometry_file, dl2_proton_geometry_file
+):
+    from ctapipe.tools.train_energy_regressor import TrainEnergyRegressor
+    from ctapipe.tools.train_particle_classifier import TrainParticleClassifier
+
+    tool = TrainEnergyRegressor()
+    config = resource_file("ml_config.yaml")
+    out_file = model_tmp_path / "energy.pkl"
+
+    with pytest.raises(ValueError, match="Too few events"):
+        run_tool(
+            tool,
+            argv=[
+                f"--input={dl2_shower_geometry_file}",
+                f"--output={out_file}",
+                f"--config={config}",
+                "--log-level=INFO",
+            ],
+            raises=True,
+        )
+
+    tool = TrainParticleClassifier()
+    out_file = model_tmp_path / "particle_classifier.pkl"
+
+    with pytest.raises(ValueError, match="Only one class"):
+        run_tool(
+            tool,
+            argv=[
+                f"--signal={dl2_shower_geometry_file}",
+                f"--background={dl2_proton_geometry_file}",
+                f"--output={out_file}",
+                f"--config={config}",
+                "--log-level=INFO",
+            ],
+            raises=True,
+        )
+
+
+def test_cross_validation_results(model_tmp_path):
+    from ctapipe.tools.train_energy_regressor import TrainEnergyRegressor
+    from ctapipe.tools.train_particle_classifier import TrainParticleClassifier
+
+    tool = TrainEnergyRegressor()
+    config = resource_file("ml_config.yaml")
+    out_file = model_tmp_path / "energy_.pkl"
+    energy_cv_out_file = model_tmp_path / "energy_cv_results.h5"
+
+    ret = run_tool(
+        tool,
+        argv=[
+            "--input=dataset://gamma_diffuse_dl2_train_small.dl2.h5",
+            f"--output={out_file}",
+            f"--config={config}",
+            f"--cv-output={energy_cv_out_file}",
+            "--log-level=INFO",
```

Comment:
That has no use here aynmore right?

Reply:
It provides valuable info in case the test fails I think (stderror is printed nicely by pytest for failing tests)